README.md

```diff
@@ -1,49 +1,114 @@
-# setup-gap V2
+# setup-gap V3
 
 This GitHub action downloads and prepares an instance of GAP.
 It is intended to be used by the Continuous Integration (CI) action of a GAP
 package, that is by an action which runs a package's test suite.
 
 ## Supported OSes
 
-This action can be run on macOS and Ubuntu.
+This action can be run on macOS, Ubuntu and Windows (when preceded by the `setup-cygwin` action).
 
 
 ## Usage
 
 The action `setup-gap` has to be called by the workflow of a GAP
 package.
 By default it
-- downloads and compiles the master branch of GAP,
-- downloads the packages distributed with GAP, and
-- compiles the packages `io` and `profiling`
+- downloads and compiles the latest release of GAP, and
+- compiles the packages `io`, `json` and `profiling`
 
 Its behaviour can be customized via the inputs below.
 
 ### Inputs
 
 All of the following inputs are optional.
 
-- `GAP_PKGS_TO_CLONE`:
-   - A space-separated list of the GAP packages to clone.
+- `gap-version`:
+   - The gap version or branch to build. You may specify "latest" for the latest release, or "default" for the default branch.
+   - default: `latest`
+- `repository`
+   - The GitHub repository from which to clone GAP.
+   - default: `'gap-system/gap'`
+- `configflags`:
+   - Arguments to pass to the GAP configure script.
    - default: `''`
-   - example: `'io autodoc'`
-- `GAP_PKGS_TO_BUILD`:
-   - A space-separated list of the GAP packages to build. Must include
-     `io` and `profiling`.
-   - default: `'io profiling'`
-- `GAPBRANCH`:
-   - The gap branch to clone.
-   - default: `master`
-- `HPCGAP`:
-   - Build HPC-GAP if set to `yes`.
-   - default: `no`
-- `ABI`:
-   - Set to `32` to use 32bit build flags for the package
-   - default: `''`
-- `GAP_BOOTSTRAP`
-   - Which packages to build GAP with (options: full or minimal)
-   - default: `'full'`
+- `gap-pkgs-to-build`:
+   - A space-separated list of the GAP packages to build.
+   - default: `'io json profiling'`
+
+### What's new in v3
+Version v3 contains many changes compared to version v2. Simply replacing `setup-gap@v2` by `setup-gap@v3` in an existing workflow
+will almost surely not work.
```

Comment:
That sounds rather dark and ominous, without being helpful: we should at least give a hint what *needs* to be changed, and what *ought* to be changed (e.g replacing `stable-v4.X` by ... ugh... `4.X`? Or `v4.X`? Which one is allowed? If both: is there any difference?)

Reply:
I've tried to make it sound less disastrous, and indeed suggested changing branches by releases.